### PR TITLE
Parallel proving of transactions

### PR DESCRIPTION
**Pull request by MartinOndejka**
This PR can be split into 3 separate efforts.
1. Parallelising merging of transactions. This is achieved by `parallel_merger` library.
2. Distributing proving across multiple provers. This is achieved by `zeko_prover` library.
3. Integrating both into the sequencer.

Specs:
- `parallel_merger`: https://github.com/zeko-labs/zeko/blob/MartinOndejka/paralel-proving/src/app/zeko/parallel_merger/README.md
- `zeko_prover`: https://github.com/zeko-labs/zeko/blob/MartinOndejka/paralel-proving/src/app/zeko/sequencer/prover/README.md

### CODE REVIEW

**Comment by L-as** [comment]
As discussed, please add document explaining design

**Comment by MartinOndejka** [comment]
> As discussed, please add document explaining design

Added specs into the PR description.

File: src/app/zeko/da_layer/lib/diff.ml [hunk 1]
```diff
@@ -2,8 +2,39 @@ open Core_kernel
 open Mina_base
 open Mina_ledger
 
+module Without_timestamp = struct
```
Comment (L-as) [inline src/app/zeko/da_layer/lib/diff.ml:5]: Why is DA changing

**Review by L-as — APPROVED** [review]
Thanks. Of course the code isn’t perfect but we can make it better later.

**Pull request merged**